### PR TITLE
Save memory

### PR DESCRIPTION
**Pull request by bkueng**
Save memory by moving the KHP matrix to the stack and using a smaller timeout data type in DataValidator.

@priseborough @tumbili can you please review?

### CODE REVIEW

File: validation/data_validator_group.cpp [hunk 1]
```diff
@@ -49,29 +50,47 @@ DataValidatorGroup::DataValidatorGroup(unsigned siblings) :
 	_first_failover_time(0),
 	_toggle_count(0)
 {
+	assert(siblings > 0);
```
Comment (LorenzMeier) [inline validation/data_validator_group.cpp]: This will crash the drone. It should rather fail gracefully.

Comment (bkueng) [inline validation/data_validator_group.cpp]: No, you're right it's at runtime (but only debug mode). I removed it now.

**Comment by priseborough** [comment]
If you want to save some more, then the same thing can be done for KH

**Comment by bkueng** [comment]
You're right, thanks! At first it looked like all elements of KH are read, but only a few columns were set before. Now I see only the columns are used which were set right before.

**Pull request merged**